src/back/spv/mod.rs

```diff
@@ -726,8 +736,17 @@ pub fn write_vec(
     options: &Options,
     pipeline_options: Option<&PipelineOptions>,
 ) -> Result<Vec<u32>, Error> {
-    let mut words = Vec::new();
+    let mut words: Vec<u32> = Vec::new();
     let mut w = Writer::new(options)?;
-    w.write(module, info, pipeline_options, &mut words)?;
+
+    if options.flags.contains(WriterFlags::DEBUG) {}
```

Comment:
@wicast This empty `if` looks like a bug

Reply:
> @wicast This empty `if` looks like a bug

I've made another PR to fix it, alone with another debug option fix .